ServerCore/Helpers/MailHelper.cs

```diff
@@ -192,16 +210,39 @@ private static void AddRecipients(MailAddressCollection collection, IEnumerable<
             string[] addresses = recipient.Split(EmailSeparators);
             foreach (string address in addresses)
             {
-                try
-                {
-                    collection.Add(new MailAddress(address.Trim()));
-                }
-                catch (FormatException)
+                string a = address.Trim();
+                if (a != "")
```

Comment:
!string.IsNullOrWhitespace(a)

Reply:
it can't be null or we'd have an NPE on the previous line. I changed it although I think it's a little less clear this way.

Reply:
oh right sorry - can that previous line become address?.Trim() so we're covered?

Reply:
I'm not sure that helps either, just in extreme paranoia mode right now.

Reply:
Yeah, I've talked myself out of needing the address?.Trim().

Reply:
The result of recipient.split never includes nulls. In fact, it's never even an empty array. If the string is empty, it's an array with one empty string in it.

Reply:
yup yup yup